testsuite/ui/navigation/step_processor.py

```diff
@@ -0,0 +1,84 @@
+import inspect
+import logging
+from collections import namedtuple
+
+from testsuite.ui.navigation.exception import NavigationStepException
+
+
+def view_class(step, dest, **kwargs):
+    """
+    Method defining rule where
+    """
+    if inspect.isclass(step.destination) and step.destination == dest:
+        _invoke_step_method(step, dest, **kwargs)
+        return True
+    return False
+
+
+def view_class_string(step, dest, **kwargs):
+    if isinstance(step.destination, str) and step.destination == dest.__class__.__name__:
+        _invoke_step_method(step, dest, **kwargs)
+        return True
+    return False
+
+
+def href(step, dest, **kwargs):
+    if step.destination.startswith('@href'):
+        logging.debug(f"Navigator: invoking step method marked with `@href` {step}")
+        step.method(dest.path)
+        return True
+    return False
+
+
+def _invoke_step_method(step, destination_view, **kwargs):
+    """
+    Alternative steps begins with "@".
+    See `step(cls, **kwargs)` for alternatives.
+
+    :param alternative_steps: list of tuples [(method name, method)]. Describes all View methods
+        decorated with @step
+    :param destination: View class
+    :return: Bool value representing if step was preformed or not
+    """
+    try:
+        logging.debug(f"Navigator: invoking method {step} that leads to {destination_view}")
+        signature = inspect.signature(step.method)
+        filtered_kwargs = {key: value for key, value in kwargs.items() if key in signature.parameters}
+        bound = signature.bind(**filtered_kwargs)
+        bound.apply_defaults()
+        step.method(*bound.args, **bound.kwargs)
+    except Exception as exc:
+        raise NavigationStepException(step.method, destination_view) from exc
+
+
+class ViewStepsProcessor:
+    """
+    This class is responsible for processing step methods on current view with respect to
+    """
+
+    def __init__(self, current_view):
+        """
+
+        """
+        self.current_view = current_view
+        View_step = namedtuple('Step', 'method destination')
```

Comment:
This seems to be a data class with only one parameter, why you don't pass the parameter instead of creating additional objects?